jni/src/jni_util.cpp

```diff
@@ -547,6 +547,34 @@ void knn_jni::JNIUtil::SetByteArrayRegion(JNIEnv *env, jbyteArray array, jsize s
     this->HasExceptionInStack(env, "Unable to set byte array region");
 }
 
+jobject knn_jni::JNIUtil::GetObjectField(JNIEnv * env, jobject obj, jfieldID fieldID) {
```

Comment:
These seem to be just wrappers over JNIEnv, is this done for testing purposes? can we avoid this maintenance?

Reply:
Yes, it is for testing purposes. 
I could not come up with a better solution to have it here.
Also I think the motivation of introducing this class is on the testing purposes as well, right?

Reply:
yeah but most of the methods here have added logic, would be nice if we can mock JNIEnv. but should be fine if its not possible. One of the reason for this file is also caching which is handled by statics in this PR

Reply:
Unfortunately, in order to let gtest mock, the target has to be a virtual class...
And JNIEnv is a struct, and defined APIs are also not virtual methods.